pkg/natdiscovery/natdiscovery_test.go

```diff
@@ -40,335 +40,406 @@ const (
 	natNotExpected = false
```

Comment:
@tpantelis , I'm getting [1] error when running unit tests in this file , any idea how to address this issue ? 

[1] 
`[1m[38;5;9mDeferCleanup must be called inside a setup or subject node[0m
[38;5;246mDeferCleanup(func() {[0m
[38;5;243m/prj/submariner/pkg/natdiscovery/natdiscovery_suite_test.go:241[0m
  You must call DeferCleanup inside a setup node (e.g. BeforeEach, BeforeSuite,
  AfterAll...) or a subject node (i.e. It).  You can't call DeferCleanup at the
  top-level or in a container node - use the After* family of setup nodes
  instead.

  [1mLearn more at:[0m
  [38;5;14m[4mhttp://onsi.github.io/ginkgo/#cleaning-up-our-cleanup-code-defercleanup[0m
`

Reply:
`newNATDiscovery` contains a call to `DeferCleanup` which can't be done within a Context node which you're doing indirectly on L106.